### ISSUE TITLE
Update to Mapbox 2.2

### ISSUE THREAD

**Issue by jscastro76** [jscastro76/threebox#146]
New Mapbox 2.0 brings the important new feature of having finally a 3D terrain, among others such as sky layers and free camera movement. 
We'll need to analyze the impact of the change to support Mapbox 2.0 as there are important changes in the Perspective Matrixes calculations, including the max pitch change.

- [x] Depth calculation
- [x] Examples

**Comment by mohong** [comment]
The model went through the wall when I update to Mapbox 2.0.

![image](https://user-images.githubusercontent.com/7702529/103185738-a1e28b80-48f8-11eb-8851-cceabfc5bcaa.png)


Here is a gif:
![6](https://user-images.githubusercontent.com/7702529/103185815-ff76d800-48f8-11eb-8a7c-72ba3df45f7f.gif)

**Comment by jscastro76** [comment]
> The model went through the wall when I update to Mapbox 2.0.
> 
> ![image](https://user-images.githubusercontent.com/7702529/103185738-a1e28b80-48f8-11eb-8851-cceabfc5bcaa.png)
> 
> Here is a gif:
> ![6](https://user-images.githubusercontent.com/7702529/103185815-ff76d800-48f8-11eb-8a7c-72ba3df45f7f.gif)

As said in one of the open issues #146, mapbox 2.0 is the first major update in 2 years, and has changed completely the way they calculate the depth including a new free camera mode. It would require some investigation first

**Comment by KarimJedda** [comment]
I think an important consideration would be as well the new licensing model for mapbox v2 and whether or not this has implications on the licensing for this library.

**Comment by jscastro76** [comment]
> I think an important consideration would be as well the new licensing model for mapbox v2 and whether or not this has implications on the licensing for this library.

Well, this library is a three.js plug-in, it doesn’t embed mapbox, so the licensing will always depend on the developer side to configure it properly with the right licensing, as it’s nowadays.

**Comment by Stefwint** [comment]
Hi @jscastro76, I'm using your library for building shadows and it is working very nice. For version 2.0 of mapbox it doesn't work yet. Are you working on this? If so, do you have an estimation of when this will be supported?

**Comment by jscastro76** [comment]
Hi @Stefwint, thanks for the feedback
Update to Mapbox 2.0 is definitely in the backlog, but not estimated yet a date for supporting it. 

The main issue is the depth calculation between Mapbox extrusions and 3D Objects, as shown above in a gif. But it's not the only one because I haven't reviewed yet all the changes in Mapbox 2.0 that also affect the style sources. The shader created to paint the buildings shadows is based on `this.map.style.sourceCaches` which is now private and return different values.

Long story short, I cannot have a estimated date for this yet, it needs much more research.

**Comment by jscastro76** [comment]
@mohong, @KarimJedda I have finally solved the depth issues with Mapbox 2.0... 2.2.0 indeed... so this will be available very soon
@Stefwint I still fighting with the shadows
